src/util/os_utils.js

```diff
@@ -571,7 +573,7 @@ function set_yum_proxy(proxy_url) {
 function _get_dns_servers_in_forwarders_file() {
     return P.resolve()
         .then(() => {
-            if (os.type() !== 'Linux') return [];
+            if (os.type() !== 'Linux' || process.env.container === "docker") return [];
```

Comment:
This condition is repeating too many times explicitly and is susceptible to typos and bugs.
I suggest to define a const flag in the module scope that describes this situation for example:
```
const LINUX_DOCKER = process.platform === 'linux' && process.env.container === 'docker';
```
Also I think `process.platform` is better to use than `os.type()`.